src/Limelight.h

```diff
@@ -154,21 +154,20 @@ typedef struct _DECODE_UNIT {
     // (happens when the frame is repeated).
     uint16_t frameHostProcessingLatency;
 
-    // Receive time of first buffer. This value uses an implementation-defined epoch,
-    // but the same epoch as enqueueTimeMs and LiGetMillis().
-    uint64_t receiveTimeMs;
+    // Receive time of first buffer in microseconds.
+    uint64_t receiveTimeUs;
 
     // Time the frame was fully assembled and queued for the video decoder to process.
     // This is also approximately the same time as the final packet was received, so
-    // enqueueTimeMs - receiveTimeMs is the time taken to receive the frame. At the
+    // enqueueTimeUs - receiveTimeUs is the time taken to receive the frame. At the
     // time the decode unit is passed to submitDecodeUnit(), the total queue delay
-    // can be calculated by LiGetMillis() - enqueueTimeMs.
-    uint64_t enqueueTimeMs;
+    // can be calculated. This value is in microseconds.
+    uint64_t enqueueTimeUs;
 
     // Presentation time in milliseconds with the epoch at the first captured frame.
     // This can be used to aid frame pacing or to drop old frames that were queued too
     // long prior to display.
-    unsigned int presentationTimeMs;
+    uint64_t presentationTimeMs;
```

Comment:
I generally try to avoid breaking API changes if possible, but if we're going to do them here (and it's probably fine), let's just go all the way and change `presentationTimeMs` -> `presentationTimeUs` too.

Reply:
I probably screwed up here. I double-checked and this value seems to only be used internally during RTP processing and FEC recovery. I left this one at milliseconds because the original value comes from the server side at `newEntry->presentationTimeMs = packet->timestamp / PTS_DIVISOR;` and seemed self-contained to the RTP code. I think reverting to unsigned int would be fine.

Sort of related side note here, I have been wondering if we ought to be using the RTP timestamps as a way to sync up the video and audio streams, which I've seen done in other programs. I think we get away with it due to the minimal buffering, but it may still cause issues. I think this would have to start at giving the [audio](https://github.com/LizardByte/Sunshine/blob/master/src/stream.cpp#L1648) and [video](https://github.com/LizardByte/Sunshine/blob/master/src/stream.cpp#L1480-L1482) packets the same timestamp source/format. We can discuss this another time, though.

Reply:
> Sort of related side note here, I have been wondering if we ought to be using the RTP timestamps as a way to sync up the video and audio streams, which I've seen done in other programs.

It's fairly common for most applications, but real-time game streaming has incredibly stringent latency requirements. Delaying video by 40ms or whatever to sync up with audio would be totally fine for something like Zoom, but would likely make Moonlight unplayable.

Reply:
> I probably screwed up here. I double-checked and this value seems to only be used internally during RTP processing and FEC recovery. I left this one at milliseconds because the original value comes from the server side at newEntry->presentationTimeMs = packet->timestamp / PTS_DIVISOR; and seemed self-contained to the RTP code. I think reverting to unsigned int would be fine.

Sorry I meant that we should convert this to microseconds and leave it as `uint64_t`. I know it currently comes out of RTP timestamps which limits the resolution of the value, but I think we should make it microseconds to be consistent with the other values and to leave room for future timestamp improvements without having to break the API again.

Reply:
> It's fairly common for most applications, but real-time game streaming has incredibly stringent latency requirements. Delaying video by 40ms or whatever to sync up with audio would be totally fine for something like Zoom, but would likely make Moonlight unplayable.

Oh yeah, video has to be the priority, so in our case I think you would want to check video/audio pts for drift and adjust the audio stream if it was ahead or behind the video. If you're ahead of the video, play silence until the video catches up, and if you're behind, skip some samples (or possibly use some time/pitch adjustment to play it faster. That's how the Bluetooth latency reduction works in macOS Game Mode for example. I will experiment more with this, since the pts values from Sunshine are different I am not sure if this is even an issue right now. I do know that audio DACs can drift by a surprising amount, having worked on audio sync for network music players  many years ago.

> Sorry I meant that we should convert this to microseconds and leave it as uint64_t. I know it currently comes out of RTP timestamps which limits the resolution of the value, but I think we should make it microseconds to be consistent with the other values and to leave room for future timestamp improvements without having to break the API again.

Roger, I am glad you said this, it's the right way to go. When backing it out I realized I had changed it because leaving it as unsigned int and doing some annoying casting just felt bad.

Reply:
Oops, it's always MSVC catching my bugs: I forgot to fix `queue->lastOosFramePresentationTimestamp` which was still 32-bit. I am not sure why this isn't a warning on other compilers... I really appreciate the appveyor builds. :)